service/consul/datacenterWatch.go

```diff
@@ -1,109 +1,229 @@
 package consul
 
 import (
+	"context"
 	"errors"
+	"sync"
 	"time"
 
 	"github.com/go-kit/kit/log"
 	"github.com/go-kit/kit/log/level"
+	"github.com/xmidt-org/argus/chrysom"
+	"github.com/xmidt-org/argus/model"
 	"github.com/xmidt-org/webpa-common/logging"
 	"github.com/xmidt-org/webpa-common/service"
 )
 
-//datacenterWatcher checks if datacenters have been updated, based on an interval
-//If they have, then the datacenterWatcher will update the Instancers dictionary
-type datacenterWatcher struct {
+//DatacenterWatcher checks if datacenters have been updated, based on an interval
+type DatacenterWatcher struct {
```

Comment:
This struct can be private
```suggestion
type datacenterWatcher struct {
```